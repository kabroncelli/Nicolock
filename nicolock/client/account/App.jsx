import React from 'react'

import SignIn from './components/SignIn'
import Register from './components/Register'
import Forgot from './components/Forgot'

const AccountApp = (props) => {
  return (
    <div className='modal-container position-login'>
      <Register />
      <SignIn />
      <Forgot />
    </div>
  )
}

export default AccountApp
