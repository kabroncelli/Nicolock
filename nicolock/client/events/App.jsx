import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import DatePicker from './components/DatePicker'
import Calendar from './components/Calendar'

class App extends React.Component {
  static propTypes = {
    date: PropTypes.object,
  }

  constructor (props) {
    super(props)
    this.state = {
      date: props.date || moment(),
    }
  }

  dateChanged = (date) => {
    this.setState({date: date})
  }

  render () {
    return (
      <div>
        <div className='row'>
          <h1 className='title'>Events &amp;<br />Training</h1>
          <DatePicker
            date={this.state.date}
            dateChange={this.dateChanged} />
        </div>
        <Calendar
          date={this.state.date}
          dateChange={this.dateChanged} />
      </div>
    )
  }
}

export default App
