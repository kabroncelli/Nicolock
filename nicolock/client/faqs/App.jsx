import React from 'react'
import PropTypes from 'prop-types'
import helpers from './utils/helpers'

const FaqSelected = (props) => {
  const { question, answer, relatedQuestions } = props.faqSelected
  return (
    <div className='faq-wrap'>
      <div className='faq-answers'>
        <div className='question'>{question}</div>
        <div className='answer'>{answer}</div>
      </div>
      <div className='related-questions'>
        <p className='paragraph'>Related Questions:</p>
        <ul className='misc-menu faq-link'>
          {relatedQuestions ? relatedQuestions.map((question, index) => (
            <li key={`relatedQuestion-${index}`} >
              <span className='link faq neutral-0' onClick={question.selectQuestion}>{question.question}</span>
            </li>
          )) : null}
        </ul>
      </div>
    </div>
  )
}

FaqSelected.propTypes = {
  faqSelected: PropTypes.shape({
    question: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired,
    relatedQuestions: PropTypes.arrayOf(
      PropTypes.shape({
        question: PropTypes.string.isRequired,
        selectQuestion: PropTypes.func.isRequired,
      })
    ),
  }).isRequired,
}

class App extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      faqList: null,
      faqSelected: null,
    }
  }

  componentWillMount () {
    helpers.getFaqs()
      .then((faqs) => {
        const newFaqList = faqs.group(item => item.category_name)
        this.setState(() => {
          return {
            originalFaqList: faqs,
            faqList: newFaqList,
          }
        }, () => {
          if (newFaqList.length > 0) {
            this.handleChange(newFaqList[0].data[0])
          }
        })
      })
  }

  handleChange = (data) => {
    this.setState((previousState) => {
      return {
        faqSelected: {
          question: data.question,
          answer: data.answer,
          relatedQuestions: previousState.originalFaqList.filter(faq => data.related.includes(faq.id)).map(faq => {
            return {
              question: faq.question,
              selectQuestion: this.handleChange.bind(this, faq),
            }
          }),
        },
      }
    })
  }

  render () {
    return (
      <div className='trifold-content'>
        <div className='trifold-middle faq col-faq'>
          <h1 className='title'>FAQs</h1>
          {!this.state.faqList
            ? <p>Loading</p>
            : this.state.faqList.map((category) => {
              return (
                <div key={category.key}>
                  <p className='paragraph'>{category.key}:</p>
                  <ul className='misc-menu faq-link'>
                    {category.data.map((question, index) => {
                      return (
                        <li key={`question-${index}`}>
                          <span className='link faq neutral-0' onClick={() => this.handleChange(question)}>{question.question}</span>
                        </li>
                      )
                    })}
                  </ul>
                </div>
              )
            })}
        </div>
        <div className='trifold-data col-faq'>
          {!this.state.faqSelected
            ? <p>Loading</p>
            : <FaqSelected faqSelected={this.state.faqSelected} />}
        </div>
      </div>
    )
  }
}

export default App
