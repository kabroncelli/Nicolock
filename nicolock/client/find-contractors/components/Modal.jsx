import React from 'react'
import PropTypes from 'prop-types'

const Modal = (props) => {
  return (
    <div className='modal-wrap is-visible'>
      <div className='modal is-map'>
        <div className='modal-view'>
          <span className='dismiss-modal' onClick={props.dismiss}>x</span>
          <div className='modal-header'>
            <h4 className='title'>{props.title}</h4>
          </div>
          <div className='modal-content'>
            <p className='paragraph'>{props.message}</p>
          </div>
        </div>
      </div>
    </div>
  )
}

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  dismiss: PropTypes.func.isRequired,
}

export default Modal
