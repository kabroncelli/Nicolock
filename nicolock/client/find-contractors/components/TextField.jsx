import React from 'react'
import PropTypes from 'prop-types'

class TextField extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      text: '',
    }
  }

  valueEntered = (e) => {
    const newText = e.target.value
    this.setState({text: newText})
    this.props.valueEntered(newText)
  }

  render () {
    return (
      <span>
        <label className='label'>{this.props.label}</label>
        <input
          type={this.props.type}
          className='input'
          onChange={this.valueEntered}
          value={this.state.text} />
      </span>
    )
  }
}

TextField.propTypes = {
  valueEntered: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string,
}

TextField.defaultProps = {
  type: 'text',
}

export default TextField
