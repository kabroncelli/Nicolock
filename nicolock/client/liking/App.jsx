import React from 'react'
import PropTypes from 'prop-types'
import helpers from './utils/helpers'

class Like extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      count: this.props.count,
    }

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (event) {
    event.preventDefault()
    if (typeof (Storage) !== 'undefined') {
      let likeStatus = localStorage.getItem('nicolock_' + this.props.type + this.props.id)
      if (!likeStatus) {
        localStorage.setItem('nicolock_' + this.props.type + this.props.id, true)
        let liked = helpers.addLike(this.props.url)
        liked
          .then((respons) => {
            this.setState({
              count: parseInt(this.state.count) + 1,
            })
          })
      }
    }
  }

  render () {
    return (
      <a className='link like-number-link' href='#' onClick={this.handleClick}>
        <svg className='icon' version='1.1' xmlns='http://www.w3.org/2000/svg' x='0px' y='0px'
          viewBox='0 0 492.719 492.719'>
          <path className='path' d='M492.719,166.008c0-73.486-59.573-133.056-133.059-133.056c-47.985,0-89.891,25.484-113.302,63.569
            c-23.408-38.085-65.332-63.569-113.316-63.569C59.556,32.952,0,92.522,0,166.008c0,40.009,17.729,75.803,45.671,100.178
            l188.545,188.553c3.22,3.22,7.587,5.029,12.142,5.029c4.555,0,8.922-1.809,12.142-5.029l188.545-188.553
            C474.988,241.811,492.719,206.017,492.719,166.008z' />
        </svg>
        <span className='like-wrap'>Like<br />
          <span className='like-count'>
            <span className='like-number'>{this.state.count}</span> Likes
          </span>
        </span>
      </a>
    )
  }
};

Like.propTypes = {
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  count: PropTypes.string.isRequired,
}

export default Like
