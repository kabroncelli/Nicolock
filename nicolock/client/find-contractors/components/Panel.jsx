import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import Cookies from 'universal-cookie'

import Modal from './Modal'
import EnvelopeIcon from '../../nico-map/components/EnvelopeIcon'
import ReceiverIcon from '../../nico-map/components/ReceiverIcon'
import helpers from '../../nico-map/utils/helpers'

class Panel extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      success: false,
    }
  }

  saveContractor = (e) => {
    e.preventDefault()
    const cookies = new Cookies()
    const csrfToken = cookies.get('csrftoken')
    helpers.saveContractor(csrfToken, this.props.id)
      .then((response) => {
        this.setState({success: true})
      })
  }

  renderCertification = (certification) => {
    return (
      <span
        key={certification}
        className='certification'>{certification}</span>
    )
  }

  dismissModal = () => {
    this.setState({success: false})
  }

  displayModal = () => {
    if (!this.state.success) {
      return
    }
    const modal = (
      <Modal
        title='Success!'
        dismiss={this.dismissModal}
        message='The contractor has been successfully saved to your list of contractors.' />
    )
    const container = document.getElementById('modal-root')
    return ReactDOM.createPortal(
      modal, container
    )
  }

  render () {
    return (
      <div className='panel'>
        <div className='heading'>{this.props.companyName}</div>
        <div>{this.props.city}, {this.props.state}</div>
        <div>{this.props.phoneNumber}</div>
        <div className='panel-footer'>
          <div>
            {this.props.authenticated &&
              <span>
                <a className='link' href='#' onClick={this.saveContractor}>Save Contact</a>
              </span>
            }
            {this.props.profileUrl &&
              <span>
                {this.props.authenticated &&
                  <span>&nbsp; | &nbsp;</span>
                }
                <a className='link' href={this.props.profileUrl}>View Profile</a>
              </span>
            }
          </div>
          <div className='row push-half--top'>
            <span>
              {this.props.email &&
                <a className='icon' href={`mailto:${this.props.email}`}>
                  <EnvelopeIcon />
                </a>
              }
              {this.props.phoneNumber &&
                <a className='icon' href={`tel:${this.props.phoneNumber}`}>
                  <ReceiverIcon />
                </a>
              }
            </span>
            <span className='certifications'>
              {this.props.certifications.map(this.renderCertification)}
            </span>
          </div>
        </div>
        {this.displayModal()}
      </div>
    )
  }
}

Panel.propTypes = {
  companyName: PropTypes.string,
  city: PropTypes.string,
  state: PropTypes.string,
  phoneNumber: PropTypes.string,
  email: PropTypes.string,
  id: PropTypes.number.isRequired,
  authenticated: PropTypes.bool,
  profileUrl: PropTypes.string,
  certifications: PropTypes.arrayOf(PropTypes.string),
}

Panel.defaultProps = {
  certifications: [],
}

export default Panel
