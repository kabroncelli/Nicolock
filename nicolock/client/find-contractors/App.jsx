import React from 'react'
import PropTypes from 'prop-types'

import Panel from './components/Panel'
import TextField from './components/TextField'
import Dropdown from '../components/Dropdown'
import mapHelpers from '../nico-map/utils/helpers'

class App extends React.Component {
  static propTypes = {
    resourceUrl: PropTypes.string.isRequired,
    authenticated: PropTypes.bool,
  }

  constructor (props) {
    super(props)

    this.state = {
      contractors: [],
      zipCode: '',
      radius: 20,
    }
  }

  getRadiusOptions () {
    return [
      5, 10, 15, 20, 25, 30, 35, 40, 45, 50,
    ]
  }

  zipCodeChanged = (value) => {
    this.setState({zipCode: value})
  }

  radiusChanged = (value) => {
    this.setState({radius: value})
  }

  getContractors = (e) => {
    e.preventDefault()
    if (this.state.zipCode !== '') {
      mapHelpers.getNearbyZipCodes(this.state.zipCode, this.state.radius)
        .then((zipCodes) => {
          mapHelpers.getItems(this.props.resourceUrl, zipCodes)
            .then((contractors) => {
              this.setState({contractors: contractors})
            })
        })
    }
  }

  renderContractor = (contractor) => {
    return (
      <Panel
        key={contractor.id}
        id={contractor.user_id}
        companyName={contractor.name}
        phoneNumber={contractor.phone}
        email={contractor.email}
        city={contractor.city}
        state={contractor.state}
        authenticated={this.props.authenticated}
        profileUrl={contractor.profile_url}
        certifications={contractor.certifications} />
    )
  }

  render () {
    return (
      <div className='trifold-content'>
        <div className='trifold-middle'>
          <h1 className='title'>Locate A Contractor</h1>
          <p className='subtitle'>
          Enter your zip code to find contractors in your area
          </p>
          <form onSubmit={this.getContractors} className='locate-contractor-form'>
            <TextField label='Enter zip' valueEntered={this.zipCodeChanged} />
            <Dropdown
              label='Radius'
              options={this.getRadiusOptions()}
              valueEntered={this.radiusChanged} />
            <div className='action-group'>
              <button
                className='btn btn-sm neutral-0'
                type='submit'>Go</button>
            </div>
          </form>
          {this.state.contractors.map(this.renderContractor)}
        </div>
      </div>
    )
  }
}

export default App
