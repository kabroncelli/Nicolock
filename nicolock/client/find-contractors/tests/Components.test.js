import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import App from '../App'
import Panel from '../components/Panel'
import TextField from '../components/TextField'
import Modal from '../components/Modal'

configure({ adapter: new Adapter() })

describe('Find contractor components render correctly', () => {
  it('Modal renders correctly', () => {
    const dismiss = jest.fn()
    const title = 'Title!'
    const message = 'Message!'
    const wrapper = shallow(
      <Modal
        dismiss={dismiss}
        title={title}
        message={message} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Panel renders correctly', () => {
    const wrapper = shallow(
      <Panel
        id={1}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('TextField renders correctly', () => {
    const fn = jest.fn()
    const wrapper = shallow(
      <TextField
        label='Test'
        valueEntered={fn}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('App renders correctly', () => {
    const wrapper = shallow(
      <App
        resourceUrl='http://nicolock.com'
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
