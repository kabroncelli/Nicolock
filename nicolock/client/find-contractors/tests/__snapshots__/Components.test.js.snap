// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find contractor components render correctly App renders correctly 1`] = `
<div
  className="trifold-content"
>
  <div
    className="trifold-middle"
  >
    <h1
      className="title"
    >
      Locate A Contractor
    </h1>
    <p
      className="subtitle"
    >
      Enter your zip code to find contractors in your area
    </p>
    <form
      className="locate-contractor-form"
      onSubmit={[Function]}
    >
      <TextField
        label="Enter zip"
        type="text"
        valueEntered={[Function]}
      />
      <Dropdown
        label="Radius"
        options={
          Array [
            5,
            10,
            15,
            20,
            25,
            30,
            35,
            40,
            45,
            50,
          ]
        }
        valueEntered={[Function]}
      />
      <div
        className="action-group"
      >
        <button
          className="btn btn-sm neutral-0"
          type="submit"
        >
          Go
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Find contractor components render correctly Modal renders correctly 1`] = `
<div
  className="modal-wrap is-visible"
>
  <div
    className="modal is-map"
  >
    <div
      className="modal-view"
    >
      <span
        className="dismiss-modal"
        onClick={[Function]}
      >
        x
      </span>
      <div
        className="modal-header"
      >
        <h4
          className="title"
        >
          Title!
        </h4>
      </div>
      <div
        className="modal-content"
      >
        <p
          className="paragraph"
        >
          Message!
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Find contractor components render correctly Panel renders correctly 1`] = `
<div
  className="panel"
>
  <div
    className="heading"
  />
  <div>
    , 
  </div>
  <div />
  <div
    className="panel-footer"
  >
    <div />
    <div
      className="row push-half--top"
    >
      <span />
      <span
        className="certifications"
      />
    </div>
  </div>
</div>
`;

exports[`Find contractor components render correctly TextField renders correctly 1`] = `
<span>
  <label
    className="label"
  >
    Test
  </label>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</span>
`;
